### PR TITLE
Add API-Readiness-Checklist.md to RM project

### LINKED ISSUE #28: Add new API readiness checklist
**Problem description**
Adding the checklist updated as per the release management agreements

**Expected action**
Add and open for review

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by tanjadegroot**
#### What type of PR is this? 

* documentation

#### What this PR does / why we need it:

This file replaces the file in Commonalities and is updated as defined in the API release process in the Release Management Wiki Wiki (see https://wiki.camaraproject.org/display/CAM/API+Release+Process#APIReleaseProcess-APIreadinesschecklist).

#### Which issue(s) this PR fixes:

Fixes #28 

#### Changelog input

```
 Creation of updated API readiness checklist, removed from Commonalities

```

### CODE REVIEW

File: documentation/API-Readiness-Checklist.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# API Readiness Checklist
+| Nr | API release assets                           | alpha | release-candidate |  public-|release |
+|    |                                              |       |                   | initial | stable |
```
Comment (hdamker) [inline documentation/API-Readiness-Checklist.md:3]: a) I suggest to add a column where APIs can add their status or how it is intended to use the table?
b) Two lines in the header seem not to work in GitHub MarkDown, the following suggestion might work but I have not tested it.

```suggestion
| Nr | API release assets                           | alpha | release-candidate |  public-release<br>initial | public-release<br> | Status for <api-name> <version> in rX.Y |
```

Comment (hdamker) [reply documentation/API-Readiness-Checklist.md:3]: api-name, version and release can also go in a section before the table.

File: documentation/API-Readiness-Checklist.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# API Readiness Checklist
```
Comment (hdamker) [inline documentation/API-Readiness-Checklist.md:1]:
```suggestion
# API Readiness Checklist

Please find explanations for the criteria in the following checklist within [API Release Process](https://wiki.camaraproject.org/x/AgAVAQ#APIReleaseProcess-APIreadinesschecklist)
```

Comment (hdamker) [reply documentation/API-Readiness-Checklist.md:1]: Could alternatively also go below the table.

File: documentation/API-Readiness-Checklist.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+|  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |
+|  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |
+|  5 | API documentation                            |   M   |         M         |    M    |    M   |
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |
+|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |
+|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |
+|  9 | Test result statement                        |   O   |         O         |    O    |    M   |
+| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   |
+| 12 | Previous public-release was certified        |   O   |         O         |    O    |    M   |
```
Comment (hdamker) [inline documentation/API-Readiness-Checklist.md:15]: Adding the missing column and center the others:
```suggestion
|----|----------------------------------------------|:-----:|:-----------------:|:-------:|:------:|---------------------|
|  1 | API definition                               |   M   |         M         |    M    |    M   |                     |
|  2 | Design guidelines from Commonalities applied |   O   |         M         |    M    |    M   |                     |
|  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |                     |
|  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |                     |
|  5 | API documentation                            |   M   |         M         |    M    |    M   |                     |
|  6 | User stories                                 |   O   |         O         |    O    |    M   |                     |
|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |                     |
|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |                     |
|  9 | Test result statement                        |   O   |         O         |    O    |    M   |                     |
| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |                     |
| 11 | Change log updated                           |   M   |         M         |    M    |    M   |                     |
| 12 | Previous public-release was certified        |   O   |         O         |    O    |    M   |                     |
```

**Review by hdamker — APPROVED** [review]
LGTM - Thanks @tanjadegroot is shows for me what we have agreed within Release Management group. 

As this Checklist is very important for all Sub Projects I will ask today in the TSC meeting to review the PR and the linked document. So would be good to keep the PR open for about a week.

**Review by bigludo7 — APPROVED** [review]
Not a blocker, but, as we can understood the 4 columns as 4 steps should we add a note explaining that it is 'normal' to have initial public-release with same value than release candidate?

**Review by FabrizioMoggio — CHANGES_REQUESTED** [review]
Is it possible to put a link to the related documentation in the Comments column. This would improve the table also providing an overall a guide to the API development

**Review by hdamker — APPROVED** [review]

File: documentation/API-Readiness-Checklist.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+|  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |      |      |
+|  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |      |      |
+|  5 | API documentation                            |   M   |         M         |    M    |    M   |      |      |
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |      |      |
+|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |      |      |
+|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |      |      |
+|  9 | Test result statement                        |   O   |         O         |    O    |    M   |      |      |
+| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |      |      |
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   |      |      |
+| 12 | Previous public-release was certified        |   O   |         O         |    O    |    M   |      |      |
```
Comment (jlurien) [inline documentation/API-Readiness-Checklist.md:18]: If enhanced test plan is not required for the initial public release, and certification of the initial public release is required to have a stable public-release, this is assuming that the test plan used for the first certification is just the basic one, not the enhanced one, is this right? Or is that the initial public release has to be certified once the complete test plan is available.

**Comment by rartych** [comment]
> Is it possible to put a link to the related documentation in the Comments column. This would improve the table also providing an overall a guide to the API development. Maybe we could first approve the table and then improve it with this links

@tanjadegroot  I think Fabrizio wanted to add links to guidelines/templates of specific assets, but providing the links to artifacts related to given API version is good idea for me.
@FabrizioMoggio The related documentation for different assets is linked in the Explanations table in the Wiki: https://wiki.camaraproject.org/display/CAM/API+Release+Process#APIReleaseProcess-APIreadinesschecklist

**Review by rartych — APPROVED** [review]
LGTM

**Comment by tanjadegroot** [comment]
> > Is it possible to put a link to the related documentation in the Comments column. This would improve the table also providing an overall a guide to the API development. Maybe we could first approve the table and then improve it with this links
> 
> @tanjadegroot I think Fabrizio wanted to add links to guidelines/templates of specific assets, but providing the links to artifacts related to given API version is good idea for me. @FabrizioMoggio The related documentation for different assets is linked in the Explanations table in the Wiki: https://wiki.camaraproject.org/display/CAM/API+Release+Process#APIReleaseProcess-APIreadinesschecklist

The link is already there in the last line of the file

**Comment by tanjadegroot** [comment]
@soadeyemo Hi Samuel, can you please review and approve the PR ?

**Review by soadeyemo — DISMISSED** [review]
Reviewed and Approved.

**Comment by tanjadegroot** [comment]
@soadeyemo - Hi Samuel, sorry,  but can you review and approve again please, I fixed some typos and explanations in the checklist.

**Review by soadeyemo — APPROVED** [review]
Reviewed and approved.

**Pull request merged**